include/nccl_ofi_math.h

```diff
@@ -5,50 +5,81 @@
 #ifndef NCCL_OFI_MATH_H_
 #define NCCL_OFI_MATH_H_
 
+#include <algorithm>
 #include <assert.h>
 #include <stddef.h>
 #include <stdint.h>
 
+
 /*
  * @brief	Returns the ceil of x/y.
  */
-#define NCCL_OFI_DIV_CEIL(x, y) ((x) == 0 ? 0 : 1 + (((x) - 1) / (y)))
+template <class T1, class T2>
+const T1 NCCL_OFI_DIV_CEIL(const T1 &x, const T2 &y)
```

Comment:
why const qualify the return value?

the signature for this really ought to be:

```
template <typename T>
constexpr T NCCL_OFI_DIV_CEIL(T x, T y)
```

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in